packages/next/src/server/after/after-context.test.ts

```diff
@@ -459,6 +451,47 @@ describe('AfterContext', () => {
     expect(waitUntil).not.toHaveBeenCalled()
     expect(afterCallback1).not.toHaveBeenCalled()
   })
+
+  it('shadows requestAsyncStorage within after callbacks', async () => {
```

Comment:
not strictly related to this change, but we were missing a test for this. may change in the future but this is what we do for now